### ISSUE TITLE
Build Error: .eslintrc.js is not available

### ISSUE THREAD

**Issue by mattpen** [phetsims/perennial#126]
There have been two build failures due to a missing `.eslintrc.js` file in chipper this weekend.  This file is referenced by phetmarks/package.json.  In the example I posted, this file does not exist in the chipper sha mentioned.

```
Build failure: Build aborted,Error: grunt build-for-server --brand=phet --locales=lt --allHTML in ../gravity-and-orbits failed with exit code 6 and stdout:
Running "eslint:allFiles" (eslint) task
Warning: Cannot read config file: /data/share/phet/phet-repos/chipper/eslint/.eslintrc.js
Error: Cannot find module '/data/share/phet/phet-repos/chipper/eslint/.eslintrc.js'
Referenced from: /data/share/phet/phet-repos/phetmarks/package.json  Use --force to continue.

Aborted due to warnings.
. Sim = gravity-and-orbits Version = 1.1.7 Locales = lt Shas = {"comment":"# gravity-and-orbits 1.1.7 Fri Feb 08 2019 02:51:07 GMT-0700 (MST)","assert":{"sha":"a707328cc11048787a807714496af23e0bd25be1","branch":"HEAD"},"axon":{"sha":"03e252e3f7252c9853e6080ab9b2cf5369f50580","branch":"HEAD"},"brand":{"sha":"517282e9244ae8eb2b195f1927aa02e6f7255805","branch":"HEAD"},"chipper":{"sha":"1ef1836916d97b7e6977190d5bd399e4d6335103","branch":"HEAD"},"dot":{"sha":"9b3b950fd8f8a5c1efcef537ea066db7b23173d6","branch":"HEAD"},"gravity-and-orbits":{"sha":"ff3bb91cb3ae129dc4e66800b8d06565b8c79c78","branch":"HEAD"},"joist":{"sha":"08b3a9251533f730e8a9544343f5ca6d6cf1cb6c","branch":"HEAD"},"kite":{"sha":"f4c67cff6e24693f66b58fcea16069be3dbd8e88","branch":"HEAD"},"phet-core":{"sha":"c5c6c2a8273b9bd3f3bcb4ddfbcf24058b58954a","branch":"HEAD"},"phetcommon":{"sha":"fe5a0caf8972a0bebef9a55dc9ea5c973473b123","branch":"HEAD"},"query-string-machine":{"sha":"d8a4ff18a12aff003e4896213c1dfed5a79c1a30","branch":"HEAD"},"scenery":{"sha":"70ca9547aa3aa19dff9cdb20463fadb80f92ce79","branch":"HEAD"},"scenery-phet":{"sha":"9233aff82ac8e9366d444e72d40603e59a261493","branch":"HEAD"},"sherpa":{"sha":"985665662c9862d5b38583a715896029b46feb44","branch":"HEAD"},"sun":{"sha":"9c5de75906c0fa05d8535b83d394bc41374f5851","branch":"HEAD"},"tandem":{"sha":"4ac1125441b04a890088f9704314879aba5dead0","branch":"HEAD"}}
```

I'm not sure who is best assignee as I have no experience with phetmarks or eslint.  @samreid @zepumph would you be able to help?

**Comment by samreid** [comment]
I was able to reproduce the problem by checking out 1.1 branch of gravity-and-orbits and running `grunt --brands=phet`.

This could be related to https://github.com/phetsims/tasks/issues/972 (new lint structure) or perhaps https://github.com/phetsims/chipper/issues/720 (adding phetmarks to shas).  I'm not sure how to proceed, but I'm leaving @zepumph assigned in case it is the former, and adding @Denz1994 and @jonathanolson in case it is the latter.

**Comment by Denz1994** [comment]
Just noting here: I wasn't able to reproduce this build failure. 

I did a `npm update` and `npm prune` in `chipper`. Then ran `grunt checkout-target --repo=gravity-and-orbits --target=1.1` from perennial. From there `grunt --brands=phet` in the sim repo didn't produce an error.

**Comment by zepumph** [comment]
**TLDR**: It you every have this problem and just want it to be fixed so you can build:
* Create `chipper/eslint/.eslintrc.js`
* the file should look like: `module.exports = {};`

This is happening because when overhauling the eslint system to be based on a tree instead of single file, we added a pointer from each repo's package.json to a central file in chipper called `.eslintrc.js`. For older repos, they won't have that file. Basically a way to think about it is to think about how we could solve this solution by adding every single repo that gets linted as a dependency for builds. This would work because then you would check out your old version of {{insert random unversioned repo here that has a pointer to chipper/.eslintrc.js in its package.json}} and it wouldn't have that in its package.json.

Needless to say I don't like that solution. Instead I like the idea of maintenance releasing all broken branches to just have an empty `.eslintrc.js` file. I tried this out in FAMB 2.1-phetio while working on https://github.com/phetsims/phet-io/issues/1447 and it worked really well. 

Where I'm a bit confused is that, at least in that branch, we are running `eslint:allFiles`, which is part of the eslint grunt task (old code). I'm unsure why there is a descrepancy between the files it chooses to lint, and our checked dependency list. In order for this bug to occur, a file being linted would need to be in a repo that is on master (therefore pointing to a chipper file that doesn't exist). Unfortunately, we don't have a lot of control/debug capabilities when trying to dive into how eslint:allFiles is handled.

**Comment by zepumph** [comment]
The above commit worked fix the error breaking the lint task of the build. I think that we can use it whenever we run into this in the future, to cherry-pick this into the proper chipper branch. 

@jonathanolson mentioned that he thought he fixed this with other work, but I don't see any commits linked above, and it didn't seem to be fixed when I was working in FAMB 2.1-phetio. 

From here I'm not totally sure where to go. I think I will wait to see what happens in with https://github.com/phetsims/brand/issues/34, perhaps there will be a more general fix that comes out of that.

**Comment by zepumph** [comment]
@jonathanolson was a maintenance release published to handle this issue? I don't think @mattpen and I encountered this during the gzip maintenance release. Should this be closed?

**Comment by mattpen** [comment]
I haven't seen this cause any recent build failures from translations either.

**Comment by jonathanolson** [comment]
> @jonathanolson was a maintenance release published to handle this issue? I don't think @mattpen and I encountered this during the gzip maintenance release. Should this be closed?

I haven't encountered this, and I believe we are now NOT linting to prevent these types of failures. Closing!
